src/index.ts

```diff
@@ -144,4 +146,36 @@ export function snakeCase<T extends string | readonly string[]>(str?: T) {
   return kebabCase(str || "", "_") as SnakeCase<T>;
 }
 
+export function flatCase(): "";
+export function flatCase<T extends string | readonly string[]>(
+  str: T,
+): FlatCase<T>;
+export function flatCase<T extends string | readonly string[]>(str?: T) {
+  return kebabCase(str || "", "") as FlatCase<T>;
+}
+
+export function trainCase(): "";
+export function trainCase<
+  T extends string | readonly string[],
+  UserCaseOptions extends CaseOptions = CaseOptions,
+>(str: T, opts?: UserCaseOptions): TrainCase<T, UserCaseOptions["normalize"]>;
+export function trainCase<
+  T extends string | readonly string[],
+  UserCaseOptions extends CaseOptions = CaseOptions,
+>(str?: T, opts?: UserCaseOptions) {
+  return str
+    ? ((
+        (Array.isArray(str) ? str : splitByCase(str as string)) as Array<string>
+      )
+        // eslint-disable-next-line unicorn/no-array-reduce -- map and filter would require iterating twice
```

Comment:
perf: sometimes, good old fashioned for loops can do the trick and save the cost of inner function creations :)

Reply:
Definitely. For or for..of would be the most performant option for all of these, but I didn’t want to deviate too much from the general approach of the other utils.
I can of course make this a for loop, but I don’t think it will have any significant impact unless the strings are huge.

Reply:
Mainly performance overhead overhead if this util is used in critical paths (being called many times per second) and reduce causes too many memory allocations. I think not using it might be better.

Reply:
I updated the implementation with for..of. If performance is a concern here, we should also consider updating the other maps to for..of/for loops, as map is even slower than reduce.

Reply:
Thanks. Mainly wanted to avoid `reduce`. See https://github.com/unjs/scule/pull/68/commits/2535ce0382cbadf3c4c0961b87b628878c0f80db it helps to keep code consistent. Later we shall do an overall performance audit things like https://github.com/unjs/scule/issues/70 particularly can help to decide how we make pipelines on top of splitter to be most efficient.